pysmt/cmd/install.py

```diff
@@ -547,46 +153,54 @@ def print_welcome():
 def main():
     options = parse_options()
 
-    # If checking only the installation, we exit immediately
-    if options.check:
-        check_install()
-        exit(0)
-
-    if not options.skip_intro:
-        print_welcome()
-
-    # create install folder if needed
-    if not os.path.exists(BASE_DIR):
-        os.mkdir(BASE_DIR)
-
-    if options.z3:
-        install_z3(options)
+    # Address of a mirror website containing packages to avoid continuous
+    # downloads from original websites in CI
+    mirror_url = os.environ.get('PYSMT_INSTALL_MIRROR')
 
-    if options.cvc4:
-        install_cvc4(options)
+    # Env variable controlling the solvers to be installed or checked
+    requested_solvers = get_requested_solvers()
 
-    if options.msat:
-        install_msat(options)
 
-    if options.yices:
-        install_yices(options)
+    solvers_to_install = []
+    all_solvers = options.all_solvers
+    for i in INSTALLERS:
+        name = i.InstallerClass.SOLVER
+        if all_solvers or getattr(options, name) or name in requested_solvers:
+            solvers_to_install.append(i)
 
-    if options.cudd:
-        install_pycudd(options)
-
-    if options.picosat:
-        install_picosat(options)
-
-    if options.btor:
-        install_boolector(options)
+    if options.check:
+        check_installed([x.InstallerClass.SOLVER for x in solvers_to_install])
+        exit(0)
 
-    print("\n")
-    print("*" * 80)
-    print("Add the following to your .bashrc file or to your environment:")
-    print("export PYTHONPATH=\"$PYTHONPATH:"+ ":".join(PATHS) + "\"")
+    elif options.env:
+        bindings_dir= os.path.expanduser(options.bindings_path)
+        print("export PYTHONPATH=\""+ bindings_dir + ":${PYTHONPATH}\"")
 
-    with open(os.path.join(BASE_DIR, "set_paths.sh"), "a") as fout:
-        fout.write("export PYTHONPATH=\"$PYTHONPATH:"+ ":".join(PATHS) + "\"\n")
+    else:
+        # Do the actual install
+        # This should work on any platform
+        install_dir= os.path.expanduser(options.install_path)
+        if not os.path.exists(install_dir):
+            os.mkdir(install_dir)
+
+        # This should work on any platform
+        bindings_dir= os.path.expanduser(options.bindings_path)
+        if not os.path.exists(bindings_dir):
+            os.mkdir(bindings_dir)
+
+        for i in solvers_to_install:
+            installer = i.InstallerClass(install_dir=install_dir,
+                                         bindings_dir=bindings_dir,
+                                         solver_version=i.version,
+                                         mirror_link=mirror_url,
+                                         **i.extra_params)
+
+            if not installer.is_installed():
+                if not options.skip_intro:
+                    print_welcome()
+                    options.skip_intro = True
```

Comment:
It is done only once, (```skip_intro``` is set to true the first loop), but it is there to be executed only if a solver gets instaled (if it is found no message is printed).